src/user_data/mod.rs

```diff
@@ -61,6 +65,26 @@ impl Default for DataRecords {
         DataRecords::new()
     }
 }
+#[cfg(feature = "serde_support")]
+impl serde::Serialize for StatusField {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_u8(self.bits())
+    }
+}
+#[cfg(feature = "serde_support")]
+impl<'de> serde::Deserialize<'de> for StatusField {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let bits = u8::deserialize(deserializer)?;
+        StatusField::from_bits(bits)
+            .ok_or_else(|| serde::de::Error::custom("Invalid bits for StatusField"))
+    }
+}
```

Comment:
You can derive these with cfg_attr in the bitflags macro.

Reply:
somehow this was not working

Reply:
<img width="709" alt="Screenshot 2024-06-12 at 21 44 00" src="https://github.com/maebli/m-bus-parser/assets/1138612/2adb73ed-be4c-4a81-8c70-09de3a8eab76">

Reply:
what error does it show? I am doing this cargo check and it shows no error.

Reply:
oh, you also need to add bitflags/serde to the serde feature flag

Reply:
made a pr with this https://github.com/maebli/m-bus-parser/pull/11